diff -Naur fnotify.pl patch/fnotify.pl
--- fnotify.pl	2009-08-25 04:10:40.000000000 +0900
+++ patch/fnotify.pl	2009-08-25 04:10:31.000000000 +0900
@@ -47,6 +47,22 @@
 }
 
 #--------------------------------------------------------------------
+# Erase hilight's
+#--------------------------------------------------------------------
+
+sub dehilight {
+    my $unread = 0;
+    foreach (Irssi::windows()) {
+        if ($_->{data_level} == 3) {
+            $unread = 1;
+        }
+    }
+    if ($unread == 0) {
+        system("rm -f $ENV{HOME}/.irssi/fnotify");
+    }
+}
+
+#--------------------------------------------------------------------
 # The actual printing
 #--------------------------------------------------------------------
 
@@ -64,5 +80,6 @@
 
 Irssi::signal_add_last("message private", "priv_msg");
 Irssi::signal_add_last("print text", "hilight");
+Irssi::signal_add_last('window changed', 'dehilight');
 
 #- end
